File: receipt_reader/lib/utils/urls.dart
// utils/urls.dart

const int port = 2137;

class Urls {
  static String host = 'localhost';

  static Uri get basePath => Uri(
        scheme: 'http',
        host: host,
      );

  static Uri get receiveReceiptsUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/receipts/',
      );

  static Uri receiveSpecificReceiptUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: '/receipt/$id/',
      );

  static Uri get createReceiptUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/receipt/create/',
      );

  static Uri updateReceiptUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: '/receipt/update/$id/',
      );

  static Uri deleteReceiptUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: '/receipt/delete/$id/',
      );

  static Uri get registerPageUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/register/',
      );

  static Uri get loginPageUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/login/',
      );

  static Uri get logOutPageUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/logout/',
      );

  static Uri get refreshTokenUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/token/refresh/',
      );

  static Uri getUserDataUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: '/user/$id/',
      );

  static Uri get getUserSummaryUrl => Uri(
        scheme: 'http',
        host: host,
        path: '/user-summary/',
      );
  static Uri getCategoryProductsUrl(String category) => Uri(
        scheme: 'http',
        host: host,
        path: '/products/category/$category/',
      );
  static Uri changeProductUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: 'product/$id/',
      );
  static Uri get changePasswordUrl => Uri(
        scheme: 'http',
        host: host,
        path: 'user/change-password/',
      );
  static Uri getReceiptProductsUrl(int id) => Uri(
        scheme: 'http',
        host: host,
        path: 'products/$id/',
      );
}

File: backend/receiptreader/urls.py
#receiptreader/urls.py
from django.urls import path

from . import views
from .views import ChangePasswordView, ProductDetailView, ProductsByCategoryView, ProductsByReceiptView, RegisterView, LoginView, ReceiptListView, ReceiptDetailView, ShowReceiptImage, UserListView, UserDetailView, ReceiptCreateView, UpdateReceiptView, DeleteReceiptView, LogoutAPIView, UserSummaryView
from rest_framework_simplejwt.views import TokenRefreshView

urlpatterns = [
    path('', views.get_routes, name='get-routes'),
    path('token/refresh/', TokenRefreshView.as_view()),
    path('register/', RegisterView.as_view(), name='register'),
    path('login/', LoginView.as_view(), name='login'),
    path('logout/', LogoutAPIView.as_view(), name='logout'),
    path('receipts/', ReceiptListView.as_view(), name='receipt-list'),
    path('receipt/<int:pk>/', ReceiptDetailView.as_view(), name='receipt-detail'),
    path('receipt/create/', ReceiptCreateView.as_view(), name='receipt-create'),
    path('receipt/update/<int:pk>/', UpdateReceiptView.as_view(), name='receipt-update'),
    path('receipt/delete/<int:pk>/', DeleteReceiptView.as_view(), name='receipt-delete'),
    path('receipts/<int:pk>/image/<str:image_type>/<str:filename>/', ShowReceiptImage.as_view(), name='receipt-image'),
    path('users/', UserListView.as_view(), name='user-list'),
    path('user/<int:pk>/', UserDetailView.as_view(), name='user-detail'),
    path('user-summary/', UserSummaryView.as_view(), name='user-summary'),
    path('user/change-password/', ChangePasswordView.as_view(), name='change-password'),
    path('products/category/<str:category>/', ProductsByCategoryView.as_view(), name='products-by-category'),
    path('product/<int:pk>/', ProductDetailView.as_view(), name='product-detail'),
    path('products/<int:receipt_id>/', ProductsByReceiptView.as_view(), name='products-by-receipt'),
]